src/compile/decode.js

```diff
@@ -205,11 +205,25 @@ var defaultValue = function (f, def) {
 
   switch (f.type) {
     case 'string':
-      return def != null ? def : ''
+      if (def) {
+        return def
+      }
+
+      if (f.required) {
+        return ''
+      }
 
+      return null
     case 'bool':
-      return def === 'true'
+      if (def) {
+        return def === 'true'
+      }
+
+      if (f.required) {
+        return false
```

Comment:
If required, isn't the field _invalid_ in this case?

Reply:
Fair point, we are decoding here so everything that's required should have a value.